src/fee_bump_transaction.js

```diff
@@ -0,0 +1,255 @@
+import map from 'lodash/map';
+import each from 'lodash/each';
+import isString from 'lodash/isString';
+import xdr from './generated/stellar-xdr_generated';
+import { hash } from './hashing';
+
+import { StrKey } from './strkey';
+import { Keypair } from './keypair';
+import { Transaction } from './transaction';
+
+/**
+ * Use {@link TransactionBuilder.buildFeeBumpTransaction} to build a FeeBumpTransaction object, unless you have
+ * an object or base64-encoded string of the transaction envelope XDR.
+ * Once a FeeBumpTransaction has been created, its attributes and operations
+ * should not be changed. You should only add signatures (using {@link FeeBumpTransaction#sign}) to a Transaction object before
+ * submitting to the network or forwarding on to additional signers.
+ * @constructor
+ * @param {string|xdr.TransactionEnvelope} envelope - The transaction envelope object or base64 encoded string.
+ * @param {string} networkPassphrase passphrase of the target stellar network (e.g. "Public Global Stellar Network ; September 2015").
+ */
+export class FeeBumpTransaction {
+  constructor(envelope, networkPassphrase) {
+    if (typeof envelope === 'string') {
+      const buffer = Buffer.from(envelope, 'base64');
+      envelope = xdr.TransactionEnvelope.fromXDR(buffer);
+    }
+
+    if (typeof networkPassphrase !== 'string') {
+      throw new Error(
+        `Invalid passphrase provided to Transaction: expected a string but got a ${typeof networkPassphrase}`
+      );
+    }
+    this._networkPassphrase = networkPassphrase;
+
+    const envelopeType = envelope.switch();
+    if (envelopeType !== xdr.EnvelopeType.envelopeTypeTxFeeBump()) {
+      throw new Error(
+        `Invalid TransactionEnvelope: expected an envelopeTypeTxFeeBump but received an ${envelopeType.name}.`
+      );
+    }
+
+    const txEnvelope = envelope.value();
+    const sourceAccount = txEnvelope
+      .tx()
+      .feeSource()
+      .ed25519();
+
+    // since this transaction is immutable, save the tx
+    this._tx = txEnvelope.tx();
+    this._feeSource = StrKey.encodeEd25519PublicKey(sourceAccount);
+    this._fee = this._tx.fee().toString();
+
+    const innerTxEnvelope = xdr.TransactionEnvelope.envelopeTypeTx(
+      this.tx.innerTx().v1()
+    );
+    this._innerTransaction = new Transaction(
+      innerTxEnvelope,
+      networkPassphrase
+    );
+    this.signatures = map(txEnvelope.signatures() || [], (s) => s);
+  }
+
+  get tx() {
```

Comment:
it seems like this property isn't exposed in `Transaction` should we be exposing it here?

Reply:
@tamirms in the `Transaction` constructor we do `this.tx = ` which is kind of the same and allow `tx` to be modified. By doing this we disable the setter on tx. I'll update `Transaction` to follow this same patter too.